compiler/src/dotty/tools/dotc/transform/UnusedChecks.scala

```diff
@@ -0,0 +1,102 @@
+package dotty.tools.dotc.transform
+
```

Comment:
I think it's better to implement these checks as a top-down traversal in PostTyper. It's inherently a top-down algorithm not a bottom-up one.

Reply:
I will try this in PostTyper. My first approach was in typer and I quickly realized it was a mistake because these failures influenced typing desitions. Hence decided to do it in an isolated miniphase.